krabs/krabs/schema_locator.hpp

```diff
@@ -70,13 +70,13 @@ namespace std {
         size_t operator()(const krabs::schema_key &key) const
         {
             // Shift-Add-XOR hash - good enough for the small sets we deal with
-            const char* p = (const char*)&key;
             size_t h = 2166136261;
 
-            for(auto i = 0; i < sizeof(key); ++i)
-            {
-                h ^= (h << 5) + (h >> 2) + p[i];
-            }
+            h ^= (h << 5) + (h >> 2) + std::hash<krabs::guid>()(key.provider);
```

Comment:
should we instantiate std::hash<krabs::guid>() once instead of each time here?

Reply:
should `krabs::guid` expose a `hash` member that lazily computes its `std::hash` the first time it's requested?

Reply:
Just for future ref: the hasher instance has no initialization overhead. The guid instances are short lived and I _think_ the hash is only computed once.